lib/Jsrt/ChakraCore.h

```diff
@@ -163,4 +163,346 @@ JsGetModuleHostInfo(
+///     </para>
+/// </remarks>
+/// <param name="content">Pointer to string memory.</param>
+/// <param name="length">Number of bytes within the string</param>
+/// <param name="value">.</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsCreateStringUtf16(
+        const uint16_t *content,
+        size_t length,
+        _Out_ JsValueRef *value);
+
+/// <summary>
+///     Write string value into C string buffer
+/// </summary>
+/// <remarks>
+///     <para>
+///     Experimental. We may update the name or behavior until it is stable.
+///     </para>
+/// </remarks>
+/// <param name="value">JSValueRef</param>
+/// <param name="start">start offset of buffer</param>
+/// <param name="length">length to be written</param>
+/// <param name="buffer">Pointer to buffer.</param>
+/// <param name="written">Total number of characters written</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsWriteString(
+        JsValueRef value,
+        int start,
+        int length,
+        char* buffer,
+        _Out_opt_ size_t* written);
+
+/// <summary>
+///     Write string value into Utf8 string buffer
+/// </summary>
+/// <remarks>
+///     <para>
+///     Experimental. We may update the name or behavior until it is stable.
+///     </para>
+/// </remarks>
+/// <param name="value">JSValueRef</param>
+/// <param name="buffer">Pointer to buffer.</param>
+/// <param name="bufferSize">Buffer size.</param>
+/// <param name="written">Total number of characters written</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsWriteStringUtf8(
+        JsValueRef value,
+        uint8_t* buffer,
+        size_t bufferSize,
+        _Out_opt_ size_t* length);
+
+/// <summary>
+///     Write string value into Utf16 string buffer
+/// </summary>
+/// <remarks>
+///     <para>
+///     Experimental. We may update the name or behavior until it is stable.
+///     </para>
+/// </remarks>
+/// <param name="value">JSValueRef</param>
+/// <param name="start">start offset of buffer</param>
+/// <param name="length">length to be written</param>
+/// <param name="buffer">Pointer to buffer.</param>
+/// <param name="written">Total number of characters written</param>
+/// <returns>
+///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+/// </returns>
+CHAKRA_API
+    JsWriteStringUtf16(
+        JsValueRef value,
+        int start,
+        int length,
+        uint16_t* buffer,
+        _Out_opt_ size_t* written);
+
+/// <summary>
+///     Parses a script and returns a function representing the script.
+/// </summary>
+/// <remarks>
+///     Requires an active script context.
+/// </remarks>
+/// <param name="script">The script to run.</param>
```

Comment:
May want to add something like 'may accept string or external arraybuffer' to `JsValueRef script` args here and elsewhere for clarity.